### DISCUSSION TITLE
How to format the array in expanded style?

### DISCUSSION THREAD

**Discussion by acgxv** [toml-rs/toml#587] [Q&A] [answered]
```
std = ["b/std", "general-a/std", "general-c/std", "pallet-b/std", "pallet-c/std", "primitive-a/std", "pallet-a/std", "pallet-d/std"]
```

to

```
std = [
	"b/std",
	"general-a/std",
	"general-c/std",
	"pallet-a/std",
	"pallet-b/std",
	"pallet-c/std",
	"pallet-d/std",
	"primitive-a/std",
]
```

**Comment by epage** [comment]
An API that might make this easier is [visit_mut](https://docs.rs/toml_edit/latest/toml_edit/visit_mut/index.html).

For what calls to make, I'd recommend parsing and doing a debug dump of your desired format and that should give you an idea of what parts of the API on which entries you need to edit to get the desired result.

**Comment by acgxv** [comment, marked as answer]
> An API that might make this easier is [visit_mut](https://docs.rs/toml_edit/latest/toml_edit/visit_mut/index.html).
> 
> For what calls to make, I'd recommend parsing and doing a debug dump of your desired format and that should give you an idea of what parts of the API on which entries you need to edit to get the desired result.

Thank you. However, I have discovered that this approach complicates things. Instead, I have found an alternative solution:

```rs
array.push_formatted(x.decorated("\n\t", ""));
```

**Comment by nrdxp** [comment]
honestly, this is a common enough desire (for diff friendly changes) that there should frobably just be a `fmt_multiline()` for an array type to avoid manual foibles.

For now, if you want to be absolutely certain that the entire list is formatted multiline you can do something like the following:
```rust
        array.fmt(); // <- ensure the list is in a predictable format first
        for v in array.iter_mut() {
            *v = v.to_owned().decorated("\n\t", ""); // <- decorate each existing value on it's own line
        }
        array.push_formatted(x.decorated("\n\t", ",\n")); // <- place your new value on it's own line, in this case with a trailing comma and newline
 ```
